### DISCUSSION TITLE
About web bundle in theia

### DISCUSSION THREAD

**Discussion by un4gt** [eclipse-theia/theia#13927] [Q&A] [answered]
I want't to replace theia default web bundler with other tools like rspack . But I don't know how to do it. Is there any guidelines or examples ?

**Comment by msujew** [comment]
Hey @un4gt,

I don't think you will find find any guidelines or examples for that. AFAIK no one else has done this so far. There were questions in the past (see [here](https://github.com/eclipse-theia/theia/discussions/12503)) but I don't think anything came out of it.

Note that we build our own plugin for bundling [native dependencies](https://github.com/eclipse-theia/theia/tree/master/dev-packages/native-webpack-plugin). Without it, it might be difficult to bundle the backend of your Theia app.

**Reply by un4gt** [reply]
@msujew thanks for the reply.
As the number of custom extensions increases, the “yarn bundle” time is getting longer and longer. Is there any other way to reduce the time ?

**Reply by msujew** [reply, marked as answer]
> As the number of custom extensions increases, the “yarn bundle” time is getting longer and longer. Is there any other way to reduce the time ?

I'm not sure what to recommend - webpack is simply inherently slow. You can create an issue on maybe switching to a different bundler like esbuild.

The only recommendation I can give for the development workflow is: Are you using the webpack watch mode? It recompiles consistently in <1s on my machine. It works fairly well for us right now.

**Reply by un4gt** [reply]
Thanks a lot. I realized my development workflow was wrong. Since I used the `watch` mode, it has become much faster.
